cfme/modeling/base.py

```diff
@@ -160,6 +176,31 @@ def collections(self):
 
         return EntityCollections.for_entity(self, spec)
 
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, "Details")
+        except (
+            BackupNotFoundError,
```

Comment:
I'm down to run with this for now, but I'm thinking we might have to provide a common `NotFoundException` class for collecting these in a more useful manner.